app/v2/notifications/post_notifications.py

```diff
@@ -53,6 +60,50 @@
 )
 
 
+@v2_notification_blueprint.route('/{}'.format(LETTER_TYPE), methods=['POST'])
+def post_precompiled_letter_notification():
+    if 'content' not in (request.get_json() or {}):
+        return post_notification(LETTER_TYPE)
+
+    form = validate(request.get_json(), post_precompiled_letter_request)
+
+    # Check both permission to send letters and permission to send pre-compiled PDFs
+    check_service_has_permission(LETTER_TYPE, authenticated_service.permissions)
+    check_service_has_permission(PRECOMPILED_LETTER, authenticated_service.permissions)
+
+    check_rate_limiting(authenticated_service, api_user)
+
+    template = get_precompiled_letter_template(authenticated_service.id)
+
+    form['personalisation'] = {
+        'address_line_1': form['reference']
+    }
+
+    reply_to = get_reply_to_text(LETTER_TYPE, form, template)
+
+    notification = process_letter_notification(
```

Comment:
Should do we need to do any other file validation?

I know we are assuming good path but anything easy to catch? File size maybe? I'm checking for file type on the client but double up here?